packages/atlas/src/hooks/useNftTransactions.ts

```diff
@@ -0,0 +1,70 @@
+import { useApolloClient } from '@apollo/client'
+import { useCallback } from 'react'
+
+import { GetNftDocument, GetNftQuery, GetNftQueryVariables } from '@/api/queries'
+import { useConfirmationModal } from '@/providers/confirmationModal'
+import { useJoystream } from '@/providers/joystream'
+import { useTransaction } from '@/providers/transactionManager'
+import { useAuthorizedUser } from '@/providers/user'
+
+export const useNftTransactions = () => {
+  const { activeMemberId } = useAuthorizedUser()
+  const { joystream, proxyCallback } = useJoystream()
+  const handleTransaction = useTransaction()
+  const [openModal, closeModal] = useConfirmationModal()
+  const client = useApolloClient()
+
+  const _refetchData = useCallback(
+    (id: string) => {
+      client.query<GetNftQuery, GetNftQueryVariables>({
```

Comment:
Let's add `return` here. If you don't return the promise, then it cannot be awaited and `useTransaction` won't be able to catch error if it happens